action.yml

```diff
@@ -21,11 +25,27 @@ runs:
       run: |
         echo "Executing step: Prepare Gitleaks config"
         set -euo pipefail
+        
```

Comment:
The guard on this step isn’t quite right, as it only evaluates a single input parameter.

It seems like you’ll need to split this into two separate steps to handle the logic correctly. You may also need to adjust the custom rule template to account for the input variations on being present or not.

Reply:
Thanks for highlighting this, I was in rush to roll this out and took a shortcut. Now, I've refactored by introducing a new input variable 'run_hmcts_rules' and when this is set, it validates to make sure that required inputs are supplied. Also split validation and custom config file generation into two separate steps for better clarity